PublicMapping/districtbuilder
# Add support for multiple elections

## Overview

This PR adds support for regions with election data for both the 2016 & 2020 presidential elections.

### Checklist


### Demo

Election year control for map data:
![localhost_3003_projects_659b2a51-b89e-4c61-b505-19677ec659a8 (4)](https://user-images.githubusercontent.com/4432106/121903745-99d7a600-ccf6-11eb-9de5-3cf0eb8dfa37.png)
Map tooltip:
![localhost_3003_projects_659b2a51-b89e-4c61-b505-19677ec659a8 (5)](https://user-images.githubusercontent.com/4432106/121903915-c25fa000-ccf6-11eb-9cd2-85544960be84.png)
Sidebar tooltip:
![localhost_3003_projects_659b2a51-b89e-4c61-b505-19677ec659a8 (3)](https://user-images.githubusercontent.com/4432106/121903746-9a703c80-ccf6-11eb-93c8-585aa17b134c.png)
Sidebar label tooltip:
![localhost_3003_projects_659b2a51-b89e-4c61-b505-19677ec659a8 (2)](https://user-images.githubusercontent.com/4432106/121903747-9a703c80-ccf6-11eb-9e3f-253789c2bf3a.png)
New keyboard shortcut:
![localhost_3003_projects_659b2a51-b89e-4c61-b505-19677ec659a8 (1)](https://user-images.githubusercontent.com/4432106/121903749-9b08d300-ccf6-11eb-848f-07a6a50f120d.png)
Toast for new keyboard shortcut:
![localhost_3003_projects_659b2a51-b89e-4c61-b505-19677ec659a8](https://user-images.githubusercontent.com/4432106/121903750-9b08d300-ccf6-11eb-9302-f887bc72665e.png)


### Notes

We had originally written this in a more generic fashion, and I'm choosing in this PR to hardcode in support for the following three options instead, to make the code easier to work with:
 - Unspecified election, keys are `democrat` / `republican` / `other party` (what we currently have, I assume the election is 2016)
 - Both 2016 & 2020, keys are  `democrat16` / `republican16` / `other party16` /  `democrat20` / `republican20` / `other party20` (this is what the new IL data has)
 - Just 2016, but explicitly specified by the key name (We don't have any states like this yet, but I added support as I know some upcoming states will only have 2016 data and this should make the preprocessing easier)
 
 
 I played with putting this into the type signature for `IStaticMetadata.voting` (which currently allows any `string` keys), but backed away from it when it looked like a more extensive refactor than I wanted to get into.
 
 
## Testing Instructions

- Use the `load-dev-data` script or update your database to use the `IL` data in `s3://global-districtbuilder-dev-us-east-1/regions/US/IL/2021-06-04T15:05:37.089Z/`
  - Interactions on the map page should show election data, and the sidebar should show PVI
- Also ensure you have a state in your database with pre-existing election data, either `PA` or `VA`, to test for regressions
  - You should not be able to switch election years for a state with data for only 1 election
- For a state/region with no election data at all, there should also be no regressions



Closes #808 
Closes #807
